lib/WeBWorK3/Controller/Problem.pm

```diff
@@ -4,87 +4,110 @@ use strict;
 
 use Mojo::Base 'Mojolicious::Controller', -signatures;
 
-sub addProblem ($self) {
-	my $problem = $self->schema->resultset('SetProblem')->addSetProblem(
-		params => {
-			course_id => int($self->param('course_id')),
-			set_id    => int($self->param('set_id')),
-			%{ $self->req->json }
+sub getAllProblems ($c) {
```

Comment:
This conversion of `$self` to `$c` and `$self` to `$app` is exactly the sort of thing that you should do in a dedicated pull request.  If that is all that is in the pull request, it is easy to review and would be merged quickly.  By doing this in this pull request you make it so much harder to review, and slow down the merge of this pull request.

Reply:
Yes, absolutely. But, to be clear, this is not a request to revert -- just a note for the future? These changes should be one cleanup PR, separate from a feature PR, yeah?